### DISCUSSION TITLE
How can I get the content when the comment changes?

### DISCUSSION THREAD

**Discussion by xrxrjiang** [microsoft/vscode-discussions#446] [Extension Development QnA] [answered]
When I start discussion, I want to cache the content,  how can I get the content onchange ?
<img width="677" alt="image" src="https://user-images.githubusercontent.com/23227390/217705487-5c54964a-55e5-41f3-bd94-5ea872cf4e6a.png">

**Comment by KrabsWong** [comment, marked as answer]
I have the same requirement.

In order to prevent missing comment for any reason (such as network issue, system issue, refresh page or something else), I wanna cache what user typed in the dialog. The best way, I think, is capture user's input timely and refresh the local cache, and remove it after finishing submit or cancel.

But, the same as @xrxrjiang, I can't find any api to capture user's input timely.


Update:

So, that's the solution

```typescript
vscode.workspace.onDidChangeTextDocument(listeners => {
  const { contentChanges, document } = listeners;

  // filter condition with uri scheme
  if (document.uri.scheme !== 'comment') {
    return;
  }

  // Use data from contentChanges to build the user's input content. (add/remove/change and blablabla)
  // Do what you need.
});
```

**Reply by KrabsWong** [reply]
So, that's the solution

```typescript
vscode.workspace.onDidChangeTextDocument(listeners => {
  const { contentChanges, document } = listeners;

  // filter condition with uri scheme
  if (document.uri.scheme !== 'comment') {
    return;
  }

  // Use data from contentChanges to build the user's input content. (add/remove/change and blablabla)
  // Do what you need.
});
```

Move into the first reply.

**Reply by ghost** [reply]
(@KrabsWong please edit the code block back into your issue! Then it can be accepted as answer)

**Reply by KrabsWong** [reply]
Updated, thanks.
